pkg/cluster-operator/customcluster_helper.go

```diff
@@ -452,17 +450,22 @@ func (r *CustomClusterController) findManageWorkerPod(ctx context.Context, custo
 // Kubespray v2.20.0 supports Kubernetes versions from 1.22.0 to 1.24.6,
 // while Kubespray v2.22.1 supports Kubernetes versions from 1.24.0 to 1.26.5.
 // This function returns one of these two Kubespray versions based on the input Kubernetes version.
-func getKubesprayImage(kubeVersion string) (string, error) {
+func getKubesprayImage(ctx context.Context, kubeVersion string) string {
+	log := ctrl.LoggerFrom(ctx)
 	var kubesprayVersion string
 	kubeVersion = strings.TrimPrefix(kubeVersion, "v")
 	targetVersion, err := semver.NewVersion(kubeVersion)
+	// should not happen
```

Comment:
why cannot it happen? Do we have validation on `kubeVersion`

Reply:
you are right, we should validation on it